gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/engine/VertxPluginRegistry.java

```diff
@@ -50,193 +61,172 @@
  * @author eric.wittmann@redhat.com
  */
 public class VertxPluginRegistry extends DefaultPluginRegistry {
+    private final IApimanLogger LOG = ApimanLoggerFactory.getLogger(VertxPluginRegistry.class);
+    private final Vertx vertx;
 
-    private Vertx vertx;
-    private ProxyOptions sslProxy;
-    private String sslNoProxy;
-    private ProxyOptions proxy;
-    private String noProxy;
-
+    private final InheritingHttpClientOptions defaultHttpClientOptions = new InheritingHttpClientOptions();
+    private final SysPropsProxySelector proxySelector = new SysPropsProxySelector();
+    // TODO(msavy): We could add an LRU cache here that's a combination of URL + proxy info to HttpClient?
```

Comment:
Maybe we should create an issue for that?

Reply:
Was more that if anyone complains about that, we can add it. I really think it's optional.

Reply:
i.e. this would only really be salient if someone is loading hundreds of plugins per second or something, which seem rather unlikely 🤔. 

But I felt it better to leave that there in case there was a situation I hadn't thought about and it could be done later.